async_simple/Executor.h

```diff
@@ -88,8 +88,8 @@ class Executor {
     // should guarantee that the func would be executed.
     virtual bool schedule(Func func) = 0;
     // Return true if caller runs in the executor.
-    virtual bool currentThreadInExecutor() const = 0;
-    virtual ExecutorStat stat() const = 0;
+    virtual bool currentThreadInExecutor() const { return false; }
```

Comment:
Future/Promise components used this interface to check current context, If user used Future/Promise may got unexpected error with this default implement, how about to throw the exception to tell user need to implement it?

Reply:
Good, could you give advice about throwing what kind of exception.

Reply:
std::logic_error ?

Reply:
OK